modules/core/src/system.cpp

```diff
@@ -584,6 +585,9 @@ struct HWFeatures
         have[CV_CPU_FP16] = true;
     #endif
     #endif
+    #if CV_DOTPROD
+        have[CV_CPU_DOTPROD] = true;
```

Comment:
Unfortunately, this flag has to be checked more carefully.
`CV_DOTPROD` only means that the compiler is capable of dot product instruction and the user wishes to enable this instruction.
What this flag is supposed to express is whether the CPU supports dot product instruction.

Another unfortunate thing is that till today, OpenCV only enabled NEON and FP16 instructions, which are NOT extended instructions in Aarch64, so this could be determined in compile time.
In this PR, the dot prod instruction is an extended feature, and the program has to check it before.

Some pointers
 - [`HWCAP_ASIMDDP` is what you are looking for](https://www.kernel.org/doc/html/latest/arm64/elf_hwcaps.html#the-hwcaps-exposed-in-at-hwcap )
 - [What's happening under the hood](http://tomoyo.osdn.jp/cgi-bin/lxr/source/arch/arm64/kernel/cpuinfo.c#L66)

Here's the corresponding code in OpenCV (for Armv7)
https://github.com/opencv/opencv/blob/3ff1e5e6b1193421b2556683d2a6a0edda61dc31/modules/core/src/system.cpp#L560-L578
You need to populate this part and do equivalent thing in Arm v8